hack/log/log-dump.sh

```diff
@@ -86,6 +86,15 @@ dump_mgmt_cluster_logs() {
     kind export logs --name="${mgmt_cluster_name}" "${ARTIFACTS}/management-cluster"
 }
 
+dump_kube_system_pod_describes() {
+    workload_cluster_dir=$1
+    echo "Collecting describe information of Pods in kube-system namespace"
+    mkdir -p  "${workload_cluster_dir}/describes"
```

Comment:
it's a little weird having pod logs for each node in https://gcsweb.k8s.io/gcs/kubernetes-jenkins/pr-logs/pull/kubernetes-sigs_cluster-api-provider-azure/2087/pull-cluster-api-provider-azure-ci-entrypoint/1494146202088771584/artifacts/workload-cluster/capz-na6w2g-control-plane-vpp5z/pods/, and then the describes all in one place here https://gcsweb.k8s.io/gcs/kubernetes-jenkins/pr-logs/pull/kubernetes-sigs_cluster-api-provider-azure/2087/pull-cluster-api-provider-azure-ci-entrypoint/1494146202088771584/artifacts/workload-cluster/

Maybe we could modify the log-dump-daemonset to also grab pod describes and keep the same folder structure?

Reply:
I'm not sure how to do it with `log-dump-daemonset` but I updated the script. Now the folder structure is like:
```
workload-cluster
  node-lin
    pods
      pod0
         folders/
         pod-describe
    kubelet.log
    ...
  node-win
    pods
      pod0
         folders/
         pod-describe
    kubelet.log
    ...
```
https://gcsweb.k8s.io/gcs/kubernetes-jenkins/pr-logs/pull/kubernetes-sigs_cluster-api-provider-azure/2087/pull-cluster-api-provider-azure-ci-entrypoint/1495574875039862784/artifacts/workload-cluster/capz-mxu2fy-md-0-9w2xj/pods/kube-system_calico-node-cjjbp/
https://gcsweb.k8s.io/gcs/kubernetes-jenkins/pr-logs/pull/kubernetes-sigs_cluster-api-provider-azure/2087/pull-cluster-api-provider-azure-ci-entrypoint/1495574875039862784/artifacts/workload-cluster/capz-mxu2-2kxsl/pods/kube-system_kube-proxy-windows-m8vn9/